packages/block-library/src/navigation/edit/index.js

```diff
@@ -263,7 +260,13 @@ function Navigation( {
 		 */
 		__unstableMarkNextChangeAsNotPersistent();
 		setRef( fallbackNavigationMenus[ 0 ].id );
-	}, [ navigationMenus ] );
+	}, [
+		navigationMenus,
+		ref,
+		isCreatingNavigationMenu,
+		fallbackNavigationMenus,
```

Comment:
I think `fallbackNavigationMenus` should be memoized, otherwise it'll trigger this effect on every render.

Reply:
> I think fallbackNavigationMenus should be memoized (on line 227), otherwise it'll trigger this effect on every render (since it creates a new array each render), and it could be fairly expensive.

Yes that's correct. Nice spot. I've now memoized it.

> Possibly it could also be sorted (lines 248-252) as part of that memoization.

I agreed with this and so I've added sorting into the memoized value which is recomputed if there are new navigation menus.